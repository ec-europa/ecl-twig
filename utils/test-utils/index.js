const Twig = require('twig');
const { format } = require('prettier');

const renderTwigFile = (file, options, cb) =>
  Twig.renderFile(file, options, (err, html) =>
    cb(err, format(html, { parser: 'html' }))
  );

const renderTwigFileAsNode = (file, options) =>
  new Promise((resolve, reject) =>
    Twig.renderFile(file, options, (err, html) => {
      if (err) return reject(err);

      const div = document.createElement('div');
      div.innerHTML = html.trim();

      return resolve(div.firstChild);
    })
  );

module.exports = {
  renderTwigFile,
  renderTwigFileAsNode,
};
