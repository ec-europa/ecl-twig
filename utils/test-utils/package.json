{
  "name": "@ecl-twig/test-utils",
  "version": "2.29.0",
  "description": "A local test helper",
  "main": "index.js",
  "private": true,
  "devDependencies": {
    "deepmerge": "4.2.2"
  }
}
