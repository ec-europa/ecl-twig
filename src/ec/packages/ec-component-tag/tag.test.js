import path from 'path';
import { renderTwigFileAsNode } from '@ecl-twig/test-utils';

describe('EC - Tag', () => {
  const template = path.resolve(__dirname, './tag.html.twig');
  const render = params => renderTwigFileAsNode(template, params);
  const defaultIconPath = 'static/icons.svg';
  const defaultDataStructure = {
    tag: {
      path: '',
      type: '',
      label: '',
    },
    default_icon_path: defaultIconPath,
  };

  describe('Link', () => {
    test('renders correctly', () => {
      expect.assertions(1);

      defaultDataStructure.tag.type = 'link';
      defaultDataStructure.tag.label = 'Link tag';
      defaultDataStructure.tag.path = '/example-path';

      return expect(render(defaultDataStructure)).resolves.toMatchSnapshot();
    });
  });

  describe('Button', () => {
    test('renders correctly', () => {
      expect.assertions(1);

      defaultDataStructure.tag.type = 'button';
      defaultDataStructure.tag.label = 'Button tag';

      return expect(render(defaultDataStructure)).resolves.toMatchSnapshot();
    });
  });

  describe('Removable', () => {
    test('renders correctly', () => {
      expect.assertions(1);

      defaultDataStructure.tag.type = 'removable';
      defaultDataStructure.tag.label = 'Removable tag';
      defaultDataStructure.default_icon_path = defaultIconPath;

      return expect(render(defaultDataStructure)).resolves.toMatchSnapshot();
    });
  });
});
