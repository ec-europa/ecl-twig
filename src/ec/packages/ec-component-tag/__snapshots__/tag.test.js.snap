// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EC - Tag Button renders correctly 1`] = `
<button
  class="ecl-tag"
  type="button"
>
  Button tag
</button>
`;

exports[`EC - Tag Button renders correctly with extra attributes 1`] = `
<button
  class="ecl-tag"
  data-test="data-test-value"
  data-test-1="data-test-value-1"
  type="button"
>
  Button tag
</button>
`;

exports[`EC - Tag Button renders correctly with extra class names 1`] = `
<button
  class="ecl-tag custom-class custom-class--test"
  type="button"
>
  Button tag
</button>
`;

exports[`EC - Tag Link renders correctly 1`] = `
<a
  class="ecl-tag"
  href="/example-path"
>
  Link tag
</a>
`;

exports[`EC - Tag Link renders correctly with extra attributes 1`] = `
<a
  class="ecl-tag"
  data-test="data-test-value"
  data-test-1="data-test-value-1"
  href="/example-path"
>
  Link tag
</a>
`;

exports[`EC - Tag Link renders correctly with extra class names 1`] = `
<a
  class="ecl-tag custom-class custom-class--test"
  href="/example-path"
>
  Link tag
</a>
`;

exports[`EC - Tag Removable renders correctly 1`] = `
<button
  class="ecl-tag ecl-tag--removable"
  type="button"
>
  Removable tag
  <span
    class="ecl-tag__icon"
  >
    <svg
      aria-hidden="true"
      class="ecl-icon ecl-icon--xs ecl-tag__icon-close"
      focusable="false"
    >
      <use
        xlink:href="static/icons.svg#ui--close"
      />
    </svg>
    <svg
      aria-hidden="true"
      class="ecl-icon ecl-icon--xs ecl-tag__icon-close-filled"
      focusable="false"
    >
      <use
        xlink:href="static/icons.svg#ui--close-filled"
      />
    </svg>
  </span>
</button>
`;

exports[`EC - Tag Removable renders correctly with extra attributes 1`] = `
<button
  class="ecl-tag ecl-tag--removable"
  data-test="data-test-value"
  data-test-1="data-test-value-1"
  type="button"
>
  Removable tag
  <span
    class="ecl-tag__icon"
  >
    <svg
      aria-hidden="true"
      class="ecl-icon ecl-icon--xs ecl-tag__icon-close"
      focusable="false"
    >
      <use
        xlink:href="static/icons.svg#ui--close"
      />
    </svg>
    <svg
      aria-hidden="true"
      class="ecl-icon ecl-icon--xs ecl-tag__icon-close-filled"
      focusable="false"
    >
      <use
        xlink:href="static/icons.svg#ui--close-filled"
      />
    </svg>
  </span>
</button>
`;

exports[`EC - Tag Removable renders correctly with extra class names 1`] = `
<button
  class="ecl-tag ecl-tag--removable custom-class custom-class--test"
  type="button"
>
  Removable tag
  <span
    class="ecl-tag__icon"
  >
    <svg
      aria-hidden="true"
      class="ecl-icon ecl-icon--xs ecl-tag__icon-close"
      focusable="false"
    >
      <use
        xlink:href="static/icons.svg#ui--close"
      />
    </svg>
    <svg
      aria-hidden="true"
      class="ecl-icon ecl-icon--xs ecl-tag__icon-close-filled"
      focusable="false"
    >
      <use
        xlink:href="static/icons.svg#ui--close-filled"
      />
    </svg>
  </span>
</button>
`;
